presto-main/src/main/java/com/facebook/presto/cost/ComparisonStatsCalculator.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.cost;
+
+import com.facebook.presto.sql.planner.Symbol;
+import com.facebook.presto.sql.tree.ComparisonExpressionType;
+
+import static com.facebook.presto.cost.FilterStatsCalculator.filterStatsForUnknownExpression;
+import static com.facebook.presto.cost.SymbolStatsEstimate.buildFrom;
+import static java.lang.Double.NEGATIVE_INFINITY;
+import static java.lang.Double.NaN;
+import static java.lang.Double.POSITIVE_INFINITY;
+import static java.lang.Double.isNaN;
+import static java.lang.Math.max;
+
+public final class ComparisonStatsCalculator
+{
+    private ComparisonStatsCalculator() {}
+
+    public static PlanNodeStatsEstimate comparisonSymbolToLiteralStats(PlanNodeStatsEstimate inputStatistics, Symbol symbol, double doubleLiteral, ComparisonExpressionType type)
+    {
+        switch (type) {
+            case EQUAL:
+                return symbolToLiteralEquality(inputStatistics, symbol, doubleLiteral);
+            case NOT_EQUAL:
+                return symbolToLiteralNonEquality(inputStatistics, symbol, doubleLiteral);
+            case LESS_THAN:
+            case LESS_THAN_OR_EQUAL:
+                return symbolToLiteralLessThan(inputStatistics, symbol, doubleLiteral);
+            case GREATER_THAN:
+            case GREATER_THAN_OR_EQUAL:
+                return symbolToLiteralGreaterThan(inputStatistics, symbol, doubleLiteral);
+            case IS_DISTINCT_FROM:
+            default:
+                return filterStatsForUnknownExpression(inputStatistics);
+        }
+    }
+
+    private static PlanNodeStatsEstimate symbolToLiteralRangeComparison(PlanNodeStatsEstimate inputStatistics, Symbol symbol, StatisticRange literalRange)
```

Comment:
Please explain

Reply:
I am sorry. I meant: `ince literalRange is a range, do we still care it is a literal`? Because to me it looks like we could simply name this method to `symbolToRangeComparison` and `s/literalRange/range`